services/app-api/src/resolvers/contract/updateContractDraftRevision.ts

```diff
@@ -0,0 +1,142 @@
+            logError('updateContractDraftRevision', errMessage)
+            setErrorAttributesOnActiveSpan(errMessage, span)
+            throw new UserInputError(errMessage, {
+                argumentName: 'draftRevision',
+            })
+        }
+
+        const contractWithHistory =
+            await store.findContractWithHistory(contractID)
+
+        if (contractWithHistory instanceof Error) {
+            const errMessage = `Issue finding contract message: ${contractWithHistory.message}`
+            setErrorAttributesOnActiveSpan(errMessage, span)
+
+            if (contractWithHistory instanceof NotFoundError) {
+                throw new GraphQLError(errMessage, {
+                    extensions: {
+                        code: 'NOT_FOUND',
+                        cause: 'DB_ERROR',
+                    },
+                })
+            }
+
+            throw new GraphQLError(errMessage, {
+                extensions: {
+                    code: 'INTERNAL_SERVER_ERROR',
+                    cause: 'DB_ERROR',
+                },
+            })
+        }
+
+        // This resolver is only callable by state users from the same state as the contract
+        if (
+            !isStateUser(context.user) ||
+            contractWithHistory.stateCode !== context.user.stateCode
+        ) {
+            const msg = !isStateUser(context.user)
+                ? 'User not authorized to modify state data'
+                : 'User not authorized to fetch data from a different state'
+            logError('updateContractDraftRevision', msg)
+            setErrorAttributesOnActiveSpan(msg, span)
+            throw new ForbiddenError(msg)
+        }
+
+        if (
+            !contractWithHistory.draftRevision ||
+            contractWithHistory.status === 'SUBMITTED' ||
+            contractWithHistory.status === 'RESUBMITTED'
+        ) {
+            const errMessage = `Contract is not in editable state. Contract: ${contractID} Status: ${contractWithHistory.status}`
+            logError('updateContractDraftRevision', errMessage)
+            setErrorAttributesOnActiveSpan(errMessage, span)
+            throw new UserInputError(errMessage, {
+                argumentName: 'contractID',
+            })
+        }
+
+        // If updatedAt does not match concurrent editing occurred.
+        if (
+            contractWithHistory.draftRevision.updatedAt.getTime() !==
+            draftRevision.updatedAt.getTime()
+        ) {
+            const errMessage = `Concurrent update error: The data you are trying to modify has changed since you last retrieved it. Please refresh the page to continue.`
+            logError('updateContractDraftRevision', errMessage)
+            setErrorAttributesOnActiveSpan(errMessage, span)
+            throw new UserInputError(errMessage)
+        }
+
+        // Using zod to validate and transform graphQL types into domain types.
+        const parsedFormData = validateContractDraftRevisionInput(
```

Comment:
I would have this function return our `DomainModel | Error` just like we do error handling everywhere else. I think it's reasonable to return the ZodError b/c that already has all the error paths in it that didn't validate. We might want to add some of that info into custom error sections for graphql as well, but I think starting with just the message is fine